snapcraft/internal/elf.py

```diff
@@ -376,6 +377,16 @@ def load_dependencies(
                 library_paths.add(l.path)
         return library_paths
 
+    def is_dynamic_executable(self) -> bool:
+        # This way of answering the question is perhaps a bit OTT. But it works.
+        try:
+            ef = elftools.elf.elffile.ELFFile(open(self.path, "rb"))
+        except ELFError as e:
```

Comment:
`as e` can be better expressed as `as elf_error`

Reply:
Done.